src/components/UrlUtilsService.js

```diff
@@ -48,13 +48,44 @@ goog.provide('ga_urlutils_service');
                   /.*kmz$/.test(url));
         };
 
-        this.proxifyUrl = function(url) {
+        // Test using a head request if the remote resource enables CORS
+        this.isCorsEnabled = function(url) {
+          var deferred = $q.defer();
+          $http.head(url, { timeout: 1500 }).then(function(response) {
+            // If the head request responded successfully it means that CORS
+            // is enabled
```

Comment:
Now you re right we may have false positive in case the server doesn t allow head requests. Head requests do pre-flight, like get. http://stackoverflow.com/questions/22107694/why-does-a-cross-origin-head-request-need-a-preflight-check
Best would be to simulate the option request and check the content or simply do a first get and use the proxy on error.